projects/step-frontend/src/lib/modules/timeseries/time-selection/time-range-picker.component.ts

```diff
@@ -140,12 +139,32 @@ export class TimeRangePicker implements OnInit {
 
   setFromDate(event: MatDatepickerInputEvent<any>) {
     let date = new Date(event.value.ts);
-    this.fromDateString = `${date.toLocaleDateString()} 00:00:00`;
+    date.setHours(0);
+    this.fromDateString = this.formatInputDate(date);
   }
 
   setToDate(event: MatDatepickerInputEvent<any>) {
     let date = new Date(event.value.ts);
-    this.toDateString = `${date.toLocaleDateString()} 00:00:00`;
+    date.setHours(0);
+    this.toDateString = this.formatInputDate(date);
+  }
+
+  formatInputDate(date: Date) {
+    let isoFullDate = date.toISOString();
+    let isoDate = isoFullDate.substring(0, isoFullDate.indexOf('T'));
+    let hours = this.formatTimeValue(date.getHours());
+    let minutes = this.formatTimeValue(date.getMinutes());
+    let seconds = this.formatTimeValue(date.getSeconds());
+    let isoTime = `${hours}:${minutes}:${seconds}`;
+    return `${isoDate} ${isoTime}`;
+  }
+
+  formatTimeValue(value: number) {
+    if (value < 10) {
+      return '0' + value;
+    } else {
+      return value;
```

Comment:
might be nicer as:

`String(value).padStart(2, '0');`

Reply:
I would probably directly use this function where needed since at least for me padStart is clearer then "formatTimeValue" to understand what the function does.

Reply:
it's way better. I wasn't aware of this method.